lib/plugins/aws/customResources/resources/utils.js

```diff
@@ -55,22 +55,34 @@ function response(event, context, status, data = {}, err) {
   });
 }
 
-function getLambdaArn(region, accountId, functionName) {
-  return `arn:aws:lambda:${region}:${accountId}:function:${functionName}`;
+function getLambdaArn(partition, region, accountId, functionName) {
+  return `arn:${partition}:lambda:${region}:${accountId}:function:${functionName}`;
 }
 
 function getEnvironment(context) {
   const arn = context.invokedFunctionArn.match(
-    /^arn:aws.*:lambda:(\w+-\w+-\d+):(\d+):function:(.*)$/
+    /^arn:(aws[\w-]*).*:lambda:([\w+-]{2,}\d+):(\d+):function:(.*)$/
   );
+
   return {
     LambdaArn: arn[0],
-    Region: arn[1],
-    AccountId: arn[2],
-    LambdaName: arn[3],
+    Partition: arn[1],
+    Region: arn[2],
+    AccountId: arn[3],
+    LambdaName: arn[4],
   };
 }
 
+function getPartition(region) {
+  let partition = 'aws';
+  if (region && /^cn-/.test(region)) {
+    partition = 'aws-cn';
+  } else if (region && /^us-gov-/.test(region)) {
+    partition = 'aws-us-gov';
+  }
+  return partition;
```

Comment:
There are other [less known](https://github.com/boto/botocore/blob/develop/botocore/data/endpoints.json#L5143) partitions, but this function handles 99% of use cases. I'm happy to open a new PR once this is merged to try and address the other possible partitions

Reply:
Since I can actually functionally test against `us-gov-west-1` I feel comfortable making the changes within this PR.  I'd rather not personally attempt to fix partition parsing for the entire code base (for which I am pretty unfamiliar with).